src/pudl/transform/ferc1.py

```diff
@@ -3866,3 +3885,474 @@ def plants_steam_ferc1(
         transformed_fuel=fuel_ferc1,
     )
     return convert_cols_dtypes(df, data_source="ferc1")
+
+
+##################################
+# Calculations/Metadata Management
+##################################
+
+
+class ExplodeMeta:
+    """Methods to translate XBRL metadata for preparing for explosions!"""
+
+    def __init__(self, xbrl_meta_json):
```

Comment:
unfortunately we need all the metadata in some form.

There are several things that need to happen:
* table-specific/transform param based metadata renaming (which can and should be moved to the table-transformers)
* adding  the `name_original` to each factoid's metadata (this could also be moved to the table-transformers)
* THE ONE IDK ABOUT: the `source_table` for calculation components that originate from other tables